qiita_db/processing_job.py

```diff
@@ -141,42 +141,57 @@ def create(cls, user, parameters, force=False):
         If force is True the job is gonna be created even if another job
         exists with the same parameters
         """
-        with qdb.sql_connection.TRN:
+        TTRN = qdb.sql_connection.TRN
+        with TTRN:
             command = parameters.command
 
             # check if a job with the same parameters already exists
-            sql = """SELECT processing_job_id, email, processing_job_status
+            sql = """SELECT processing_job_id, email, processing_job_status,
+                        COUNT(aopj.artifact_id)
                      FROM qiita.processing_job
                      LEFT JOIN qiita.processing_job_status
                         USING (processing_job_status_id)
+                     LEFT JOIN qiita.artifact_output_processing_job aopj
+                        USING (processing_job_id)
                      WHERE command_id = %s AND processing_job_status IN (
-                        'success', 'waiting', 'running') """
+                        'success', 'waiting', 'running') {0}
+                     GROUP BY processing_job_id, email,
+                        processing_job_status"""
 
             # we need to use ILIKE because of booleans as they can be
             # false or False
-            _format = "command_parameters->>'%s' ILIKE '%s'"
             params = []
             for k, v in viewitems(parameters.values):
                 # this is necessary in case we have an Iterable as a value
                 # but that is not unicode or string
                 if isinstance(v, Iterable) and not isinstance(v, (str,
                                                                   unicode)):
                     for vv in v:
-                        params.append(_format % (k, vv))
+                        params.extend([k, str(vv)])
                 else:
-                    params.append(_format % (k, v))
-            params = " AND ".join(params)
+                    params.extend([k, str(v)])
+
+            if params:
+                # divided by 2 as it's key-value pairs
+                len_params = len(params)/2
+                sql = sql.format(' AND ' + ' AND '.join(
+                    ["command_parameters->>%s ILIKE %s"] * len_params))
+                params = [command.id] + params
+                TTRN.add(sql, params)
+            else:
+                TTRN.add(sql.format(""), [command.id])
```

Comment:
Do you mind adding a comment here on why there is a need for a call to format with an empty string? I had to read the 3 or 4 times until I realized why it was needed.

Reply:
k, will add.